src/aind_registration_evaluation/util/file_conversion.py

```diff
@@ -0,0 +1,257 @@
+    # y_min, y_max = nonzeros[1].min(), nonzeros[1].max()
+    # z_min, z_max = nonzeros[2].min(), nonzeros[2].max()
+
+    x_min_cortical, x_max_cortical = nonzeros_cortical[0].min(), nonzeros_cortical[0].max()
+    y_min_cortical, y_max_cortical = nonzeros_cortical[1].min(), nonzeros_cortical[1].max()
+    z_min_cortical, z_max_cortical = nonzeros_cortical[2].min(), nonzeros_cortical[2].max()
+
+    # Ensure the cortical bounds do not exceed the confocal dimensions
+    x_min= x_min_cortical if x_min_cortical < data_confocal.shape[0] - 1 else data_confocal.shape[0] - 1
+    x_max = min(x_max_cortical, data_confocal.shape[0] - 1)
+    y_min= y_min_cortical if y_min_cortical < data_confocal.shape[1] - 1 else data_confocal.shape[1] - 1
+    y_max = min(y_max_cortical, data_confocal.shape[1] - 1)
+    z_min= z_min_cortical if z_min_cortical < data_confocal.shape[2] - 1 else data_confocal.shape[2] - 1
+    z_max = min(z_max_cortical, data_confocal.shape[2] - 1)
+
+    cropped_cortical = data_cortical[x_min:x_max+1, y_min:y_max+1, z_min:z_max+1]
+    cropped_confocal = data_confocal[x_min:x_max+1, y_min:y_max+1, z_min:z_max+1] 
+
+    tifffile.imwrite(outFilepath_confocal, cropped_confocal)
+    tifffile.imwrite(outFilepath_cortical, cropped_cortical)
+
+def downsample_and_int16(tiff_path, output_path, downscale_factor):
+    '''
+    Downsamples the images to a specified resolution.
+
+    Parameters
+    ------------------------
+    tiff_path: input path of TIFF image
+    output_path: desired file location of downsampled image
+    downscale_factor: 2**downscale_factor specifies the pyramid level resolution
+
+    Returns
+    ------------------------
+    None
+    '''
+
+    img_stack = tifffile.imread(tiff_path)
+
+    # Calculate new dimensions for exactly halving the current dimensions
+    new_shape = (img_stack.shape[0] // 2**downscale_factor, img_stack.shape[1] // 2**downscale_factor, img_stack.shape[2] // 2**downscale_factor)
+
+    # Resize the entire image stack using skimage
+    downsampled_img_stack = transform.resize(img_stack, new_shape, order=3, preserve_range=True).astype(np.int16)
+
+    # Save the downsampled image stack
+    tifffile.imwrite(output_path, downsampled_img_stack)
+
+def normalize_image(inFilepath, outFilepath):
+    '''
+    normalizes the image
+
+    Parameters
+        ------------------------
+        inFilepath_cortical: inFilepath of the warped image
+        outFilepath_cortical: filepath for desired filelocation of cropped cortical image
+        inFilepath_confocal: inFilepath of the target image
+        outFilepath_cortical: filepath for desired filelocation of cropped confocal image
+        threshold: intensity threshold under which the pixel will interpreted as not overlapping
+
+        Returns
+        ------------------------
+        none
+
+    '''
+    image = tifffile.imread(inFilepath)
+    normalized_image = cv2.normalize(image, None, 0, 255, cv2.NORM_MINMAX)
```

Comment:
What if you have a uint16 image? It'll fail since you're only looking in the uint8 numerical space.